validation/validate_dc2_run2.1i_object_table.ipynb

```diff
@@ -0,0 +1,733 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_four_color_color(good)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_four_color_color(stars)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_four_color_color(galaxies)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Clearly one doesn't expect the galaxies to follow the stellar locus.  The lines above are include to more easily guide the eye between the stars-only and the galaxies-only plots."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1D Density Plots"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_mag(filt, ax=None):\n",
+    "    if ax is None:\n",
+    "        ax = fig.gca()\n",
+    "    mag = f'mag_{filt}'\n",
+    "    ax.hist([good[mag], stars[mag], galaxies[mag]],\n",
+    "             label=['all', 'star', 'galaxy'],\n",
+    "             range=(16, 30),\n",
+    "             bins=np.linspace(16, 30, 100),\n",
+    "             histtype='step')\n",
+    "    ax.set_xlabel(filt)\n",
+    "    ax.set_ylabel('objects / bin')\n",
+    "    ax.legend(loc='upper left')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axes = plt.subplots(2, 3, figsize=(12, 6))\n",
+    "for ax, filt in zip(axes.flat, filters):\n",
+    "    plot_mag(filt, ax=ax)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The sharp cut in i-band is because that was the reference band for most detections.  The u-band points extend to 30th because most of them are non-detections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Blendedness and Extendedness"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w, = np.where(np.isfinite(good['blendedness']))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(len(good['blendedness']))\n",
+    "print(len(w))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "good_blendedness = good[np.isfinite(good['blendedness'])]\n",
+    "plt.hexbin(good_blendedness['mag_i'], good_blendedness['blendedness'],\n",
+    "          bins='log');\n",
+    "plt.xlabel('i')\n",
+    "plt.ylabel('blendedness');"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.hexbin(good['mag_i'], good['extendedness'],\n",
```

Comment:
The plot shows that extendedness is a binary quantity, is it worth it to also show mag_psf - mag_cModel vs mag_cModel (i.e. Figure 16 from the Bosch et al HSC paper) to remind people that extendedness, while binary, is sensitive to an ad hoc threshold choice in the delta_mag?  That might be more informative when paired with the binary extendedness plot.

Reply:
Good suggestions.  Done.